internal/util/projutil/project_util.go

```diff
@@ -283,3 +277,32 @@ func SetGoVerbose() error {
 	}
 	return nil
 }
+
+// CheckDepManagerWithRepo ensures dm and repo are being used in combination
+// correctly, as different dependency managers have different Go environment
+// requirements.
+func CheckDepManagerWithRepo(dm DepManagerType, repo string) error {
+	inGopathSrc, err := WdInGoPathSrc()
+	if err != nil {
+		return err
+	}
+	switch dm {
+	case DepManagerDep:
+		// dep assumes the project's path under $GOPATH/src is the project's
+		// repo path.
+		if repo != "" {
+			return fmt.Errorf(`repo cannot be set with dependency manager "dep"`)
```

Comment:
```suggestion
			return fmt.Errorf(`The flag --repo cannot be set with dependency manager "dep"`)
```
It's a new flag so we should make it clear.